File: python/Day1.py
def part_1() -> None:
    last: int = -1
    count: int = 0

    with open("../data/day1.txt", "r") as dFile:
        for row in dFile.readlines():
            num = int(row)

            if last > 0 and num > last:
                count += 1

            last = num

    print("Day:  1 | Part: 1 | Result:", count)


def part_2() -> None:
    nums: list[int] = list()

    with open("../data/day1.txt", "r") as dFile:
        nums = [int(row) for row in dFile.readlines()]

    last: int = -1
    count: int = 0

    for index in range(len(nums) - 2):
        num: int = sum(nums[index + off] for off in range(3))

        if last > 0 and num > last:
            count += 1

        last = num

    print("Day:  1 | Part: 2 | Result:", count)


if __name__ == "__main__":
    part_1()
    part_2()

File: go/Day1.go
package main

import (
	"aoc2021/Utils"
	"fmt"
)

func Day1_Part1() {
	nums := Utils.ReadFileInts("../data/day1.txt")

	var last = -1
	var count = 0

	for num := range nums {
		if last > 0 && nums[num] > last {
			count++
		}

		last = nums[num]
	}

	fmt.Printf("Day:  1 | Part: 1 | Result: %d\n", count)
}

func Day1_Part2() {
	nums := Utils.ReadFileInts("../data/day1.txt")

	var last = -1
	var count = 0

	for index := 0; index < len(nums)-2; index++ {
		var sum = nums[index] + nums[index+1] + nums[index+2]

		if last > 0 && sum > last {
			count++
		}

		last = sum
	}

	fmt.Printf("Day:  1 | Part: 2 | Result: %d\n", count)
}
